### PR TITLE
add missing labels to inputs

### LINKED ISSUE #3818: missing form input labels
## overview
fix some a11y issues across automate related to inputs missing labels. these were found using the [wave chrome extension](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh?hl=en-US).

## details
- [ ] compliance reports filter search needs a hidden label , there are two different inputs there that need labels
- [ ] compliance copy url input needs label
- [ ] license lockout first and last name have input label issues
- [ ] compliance profile search needs a hidden label of `Search profiles`
- [ ] data lifecycle needs hidden labels for all inputs
- [ ] attributes needs hidden label of `Search attributes`

### LINKED ISSUE #3820: form inputs not associated with labels
## overview
fix some a11y issues across automate related to inputs not being associated with their labels. these were found using the [wave chrome extension](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh?hl=en-US).

## details
- [ ] projects dropdown across automate is not associated with its label
- [ ] policies dropdown in token modal is not associated with its label

### PR DESCRIPTION
**Pull request by susanev**
Signed-off-by: susanev <susan.ra.evans@gmail.com>

### :nut_and_bolt: Description: What code changed, and why?
some inputs were missing labels, this adds them. in most cases using aria-label. this is so we can meet accessibility standards. would be rad to have some sort of automated linting for this going forward

✨  if anyone knows of other ones i missed please add commits to this pr!!!  ✨ 

Update: The second way to correctly provide accessible inputs is with the `<label for="my-id"...>` construct, where `my-id` is the actual attribute value of the associated `<input id="my-id">`. Several of the fixes herein use this approach.

### :chains: Related Resources
fixes #3818 
fixes #3820 

### :+1: Definition of Done
everything should be the same, except now folks with screenreaders can use it

### :athletic_shoe: How to Build and Test the Change
`rebuild components/automate-ui-devproxy`

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [x] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
zero.

### CODE REVIEW

**Comment by susanev** [comment]
ill prolly add in the ones described here in a #3820

**Comment by susanev** [comment]
@msorens i believe these are the valid ways
- nest the input in the label, this will automatically associate them with each other
- use a for attribute to associate the label to the input

everything needs a label, but if something happens to not have a visual label on screen, then it needs an aria-label.

if you know of more stuff feel free to push a commit!!

File: components/automate-ui/src/app/pages/automate-settings/automate-settings.component.scss [hunk 1]
```diff
@@ -24,7 +24,7 @@ h2 {
     display: flex;
 
     input {
-      width: 60px;
+      width: 64px;
```
Comment (susanev) [inline components/automate-ui/src/app/pages/automate-settings/automate-settings.component.scss:27]: bonus: noticed these inputs were not wide enough for 2 digits so increased their width by 4px

File: components/automate-ui/src/app/pages/automate-settings/automate-settings.component.html [hunk 2]
```diff
@@ -139,10 +172,17 @@ <h2>Client Runs</h2>
             </div>
           
             <p>When no Chef Infra Client run data has been received from a node in
-              <input chefInput formControlName="threshold" [resetOrigin]="shouldResetValues"
-                type="number" min="0" (keydown)="preventNegatives($event.key)" />
+              <input
+                aria-label="Amount of time before labeling a nodes as missing"
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/pages/automate-settings/automate-settings.component.html:176]: [nit]: sp - either
`aria-label="Amount of time before labeling nodes as missing"`
or
`aria-label="Amount of time before labeling a node as missing"`

Comment (susanev) [inline components/automate-ui/src/app/pages/automate-settings/automate-settings.component.html:176]:
```suggestion
                aria-label="Amount of time before labeling a node as missing"
```

**Comment by msorens** [comment]
A few others I noticed.
These I was not sure of how to handle:
https://github.com/chef/automate/blob/de85e42d1a6f29be4b36160a25a8a47b70090a2a/components/automate-ui/src/app/pages/notification-form/notification-form.component.html#L108
https://github.com/chef/automate/blob/f8ca770fc9c7915a9658855d5130811e5875eae1/components/automate-ui/src/app/pages/+compliance/+scanner/containers/nodes-edit/nodes-edit.component.html#L39
https://github.com/chef/automate/blob/f8ca770fc9c7915a9658855d5130811e5875eae1/components/automate-ui/src/app/pages/+compliance/+scanner/containers/nodes-add/nodes-add.component.html#L46

These need just "regular" fixes:
https://github.com/chef/automate/blob/de85e42d1a6f29be4b36160a25a8a47b70090a2a/components/automate-ui/src/app/pages/notification-form/notification-form.component.html#L116
https://github.com/chef/automate/blob/de85e42d1a6f29be4b36160a25a8a47b70090a2a/components/automate-ui/src/app/pages/notification-form/notification-form.component.html#L130

**Comment by susanev** [comment]
@msorens oh that notification form is getting redone, in #3708 so ill l eave those for now.

File: components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html [hunk 3]
```diff
@@ -14,32 +14,33 @@
...
         <chef-error *ngIf="credentialsLogic.showFormControlValidationError(sshForm.get('username'))">
           Username is required
         </chef-error>
       </chef-form-field>
 
       <chef-form-field>
-        <label>SSH password <span class="t-critical" aria-hidden="true">*</span></label>
-        <input chefInput formControlName="password" type="password" placeholder="password for private ssh" />
+        <label for="password">SSH password <span class="t-critical" aria-hidden="true">*</span></label>
+        <input chefInput id="password" formControlName="password" type="password" placeholder="password for private ssh" />
```
Comment (susanev) [inline components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html:34]:
```suggestion
        <label for="sshPassword">SSH password <span class="t-critical" aria-hidden="true">*</span></label>
        <input chefInput id="sshPassword" formControlName="password" type="password" placeholder="password for private ssh" />
```

File: components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html [hunk 4]
```diff
@@ -14,32 +14,33 @@
...
         <chef-error *ngIf="credentialsLogic.showFormControlValidationError(sshForm.get('name'))">
           Credential name is required
         </chef-error>
       </chef-form-field>
 
       <chef-form-field>
-        <label>SSH Username <span class="t-critical" aria-hidden="true">*</span></label>
-        <input chefInput formControlName="username" type="text" placeholder="username for private ssh" />
+        <label for="username">SSH Username <span class="t-critical" aria-hidden="true">*</span></label>
+        <input chefInput id="username" formControlName="username" type="text" placeholder="username for private ssh" />
```
Comment (susanev) [inline components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html:26]:
```suggestion
        <label for="sshUsername">SSH Username <span class="t-critical" aria-hidden="true">*</span></label>
        <input chefInput id="sshUsername" formControlName="username" type="text" placeholder="username for private ssh" />
```

File: components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html [hunk 5]
```diff
@@ -57,24 +58,24 @@
 
     <form [formGroup]="winRMForm" novalidate [ngClass]="{'credentials-form__form-error': saveErrorOccurred}">
       <chef-form-field>
-        <label>Assign a unique name for your key <span class="t-critical" aria-hidden="true">*</span></label>
-        <input chefInput formControlName="name" type="text" placeholder="friendly name for credential" />
+        <label for="name">Assign a unique name for your key <span class="t-critical" aria-hidden="true">*</span></label>
+        <input chefInput id="name" formControlName="name" type="text" placeholder="friendly name for credential" />
```
Comment (susanev) [inline components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html:62]:
```suggestion
        <label for="winrmName">Assign a unique name for your key <span class="t-critical" aria-hidden="true">*</span></label>
        <input chefInput id="winrmName" formControlName="name" type="text" placeholder="friendly name for credential" />
```

File: components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html [hunk 6]
```diff
@@ -14,32 +14,33 @@
 
     <form [formGroup]="sshForm" novalidate [ngClass]="{'credentials-form__form-error': saveErrorOccurred, 'credentials-form__form': true}">
       <chef-form-field>
-        <label>Assign a unique name for your key <span class="t-critical" aria-hidden="true">*</span></label>
-        <input chefInput formControlName="name" type="text" placeholder="friendly name for credential" />
+        <label for="name">Assign a unique name for your key <span class="t-critical" aria-hidden="true">*</span></label>
+        <input chefInput id="name" formControlName="name" type="text" placeholder="friendly name for credential" />
```
Comment (susanev) [inline components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html:18]:
```suggestion
        <label for="sshName">Assign a unique name for your key <span class="t-critical" aria-hidden="true">*</span></label>
        <input chefInput id="sshName" formControlName="name" type="text" placeholder="friendly name for credential" />
```

File: components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html [hunk 7]
```diff
@@ -57,24 +58,24 @@
...
         <chef-error *ngIf="credentialsLogic.showFormControlValidationError(winRMForm.get('name'))">
           Credential name is required
         </chef-error>
       </chef-form-field>
 
       <chef-form-field>
-        <label>WinRM username <span class="t-critical" aria-hidden="true">*</span></label>
-        <input chefInput formControlName="username" type="text" placeholder="username for WinRM" />
+        <label for="username">WinRM username <span class="t-critical" aria-hidden="true">*</span></label>
+        <input chefInput id="username" formControlName="username" type="text" placeholder="username for WinRM" />
```
Comment (susanev) [inline components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html:70]:
```suggestion
        <label for="winrmUsername">WinRM username <span class="t-critical" aria-hidden="true">*</span></label>
        <input chefInput id="winrmUsername" formControlName="username" type="text" placeholder="username for WinRM" />
```

File: components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html [hunk 8]
```diff
@@ -57,24 +58,24 @@
...
         <chef-error *ngIf="credentialsLogic.showFormControlValidationError(winRMForm.get('username'))">
           Username is required
         </chef-error>
       </chef-form-field>
 
       <chef-form-field>
-        <label>WinRM password <span class="t-critical" aria-hidden="true">*</span></label>
-        <input chefInput formControlName="password" type="password" placeholder="password for WinRM" />
+        <label for="password">WinRM password <span class="t-critical" aria-hidden="true">*</span></label>
+        <input chefInput id="password" formControlName="password" type="password" placeholder="password for WinRM" />
```
Comment (susanev) [inline components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html:78]:
```suggestion
        <label for="winrmPassword">WinRM password <span class="t-critical" aria-hidden="true">*</span></label>
        <input chefInput id="winrmPassword" formControlName="password" type="password" placeholder="password for WinRM" />
```

File: components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html [hunk 9]
```diff
@@ -90,21 +91,22 @@
 
     <form [formGroup]="sudoForm" novalidate [ngClass]="{'credentials-form__form-error': saveErrorOccurred}">
       <chef-form-field>
-        <label>Assign a unique name for your key <span class="t-critical" aria-hidden="true">*</span></label>
-        <input chefInput formControlName="name" type="text" placeholder="friendly name for credential" />
+        <label for="name">Assign a unique name for your key <span class="t-critical" aria-hidden="true">*</span></label>
+        <input chefInput id="name" formControlName="name" type="text" placeholder="friendly name for credential" />
```
Comment (susanev) [inline components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html:95]:
```suggestion
        <label for="sudoName">Assign a unique name for your key <span class="t-critical" aria-hidden="true">*</span></label>
        <input chefInput id="sudoName" formControlName="name" type="text" placeholder="friendly name for credential" />
```

File: components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html [hunk 10]
```diff
@@ -90,21 +91,22 @@
...
         <chef-error *ngIf="credentialsLogic.showFormControlValidationError(sudoForm.get('name'))">
           Credential name is required
         </chef-error>
       </chef-form-field>
 
       <chef-form-field>
-        <label>Sudo password <span class="t-critical" aria-hidden="true">*</span></label>
-        <input chefInput formControlName="password" type="password" placeholder="password for sudo" />
+        <label for="password">Sudo password <span class="t-critical" aria-hidden="true">*</span></label>
+        <input chefInput id="password" formControlName="password" type="password" placeholder="password for sudo" />
```
Comment (susanev) [inline components/automate-ui/src/app/pages/+compliance/+credentials/components/credentials-form.html:103]:
```suggestion
        <label for="sudoPassword">Sudo password <span class="t-critical" aria-hidden="true">*</span></label>
        <input chefInput id="sudoPassword" formControlName="password" type="password" placeholder="password for sudo" />
```

**Pull request merged**